### PR TITLE
Smar 2853 centralized observability for internal smartface

### PR DESCRIPTION
**Pull request by egid-fertal-inno**
The objective of this task was to create three different deployment types for the Smartface Observability Stack:

1. **Standalone Deployment**: Collects metrics, stores them, and displays the results locally.
2. **Metrics and Logs Scraper**: Scrapes metrics and logs, and sends them to a remote centralized endpoint.
3. **Centralized Aggregator Endpoint**: Serves as a central hub for storing and displaying logs and metrics from various sources.

### CODE REVIEW

File: sf-docker/sf-observability/configuration/alloy/config.alloy [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+prometheus.scrape "appmetrics" {
+	targets 		= discovery.relabel.appmetrics.output
+	forward_to      = [prometheus.remote_write.metrics_service.receiver]
+}
+
+prometheus.remote_write "metrics_service" {
+    endpoint {
+        url = sys.env("PROMETHEUS_URL")
+
+        // basic_auth {
```
Comment (Temppus) [inline sf-docker/sf-observability/configuration/alloy/config.alloy:346]: remove ?

File: sf-docker/sf-observability/configuration/grafana/datasources/datasources.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        tags: ['job', 'instance', 'pod', 'namespace']
+        mappedTags:
+          - key: 'service.name'
+            value: 'service'
+        mapTagNamesEnabled: false
+        spanStartTimeShift: '1h'
+        spanEndTimeShift: '1h'
+        filterByTraceID: false
+        filterBySpanID: false
+      #tracesToMetrics:
```
Comment (Temppus) [inline sf-docker/sf-observability/configuration/grafana/datasources/datasources.yml:29]: remove ?

File: sf-docker/sf-observability/configuration/loki/loki-config.yaml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+#
+# Statistics help us better understand how Loki is used, and they show us performance
+# levels for most users. This helps us prioritize features and documentation.
+# For more information on what's sent, look at
+# https://github.com/grafana/loki/blob/main/pkg/usagestats/stats.go
+# Refer to the buildReport method to see what goes into a report.
+#
+# If you would like to disable reporting, uncomment the following lines:
+#analytics:
+#  reporting_enabled: false
```
Comment (Temppus) [inline sf-docker/sf-observability/configuration/loki/loki-config.yaml:66]: do we want this ?

**Review by Temppus — APPROVED** [review]

File: sf-docker/sf-observability/configuration/alloy/config.alloy [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+		target_label  = "composeproject"
+		replacement   = "sf_docker"
+	}
+
+	rule {
+		source_labels = ["composeproject"]
+		regex         = "^single-camera$"
+		target_label  = "composeproject"
+		replacement   = "sf_docker"
+	}
```
Comment (FooRider) [inline sf-docker/sf-observability/configuration/alloy/config.alloy:135]: I think you can remove these rules, the "application" label is better than the "composeproject", we should throw that away.

File: sf-docker/sf-observability/configuration/alloy/config.alloy [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
```
Comment (FooRider) [inline sf-docker/sf-observability/configuration/alloy/config.alloy:261]: You can remove these label overrides as well.

File: sf-docker/sf-observability/configuration/alloy/config.alloy [hunk 6]
```diff
@@ -0,0 +1,358 @@
...
+
+	rule {
+		source_labels = ["composeproject"]
+		regex         = "^sf-docker$"
+		target_label  = "composeproject"
+		replacement   = "sf_docker"
+	}
```
Comment (FooRider) [inline sf-docker/sf-observability/configuration/alloy/config.alloy:334]: Remove these (incomplete) rules as well

File: sf-docker/sf-observability/configuration/grafana/datasources/datasources.yml [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+  - name: PostgreSQL DB
+    uid: 'provisioned-pgsql'
+    type: postgres
+    url: pgsql:5432
+    database: smartface
+    user: postgres
+    secureJsonData:
+      password: 'Test1234'
+    jsonData:
+      sslmode: 'disable'
```
Comment (FooRider) [inline sf-docker/sf-observability/configuration/grafana/datasources/datasources.yml:57]: Does any dashboard use this source?
Even if, it's not good to have it here with default password, which should be different on production installs.
Please remove this data source.

File: sf-docker/sf-observability/docker-compose.yml [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+    extends:
+      service: common
+      file: docker-compose-common.yml
+    profiles: [singlenode, multinode-aggregator]
+    image: grafana/grafana:10.1.4
+    ports:
+      - 9900:3000
+    volumes:
+      - ./configuration/grafana/datasources:/etc/grafana/provisioning/datasources
+      - ./configuration/grafana/dashboards:/etc/grafana/provisioning/dashboards
```
Comment (FooRider) [inline sf-docker/sf-observability/docker-compose.yml:51]: Please include these files using docker compose `configs`, also make sure to specify file attributes.

File: sf-docker/sf-observability/docker-compose.yml [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+    entrypoint: [ "/bin/sh","-c" ]
+    command:
+      - |
+        export HOSTNAME=$$(cat /etc/hostname)
+        /bin/alloy run --server.http.listen-addr=0.0.0.0:12345 --storage.path=/var/lib/alloy/data /etc/alloy/config.alloy
+    environment:
+      - LOKI_URL=http://loki:3100/loki/api/v1/push
+      - PROMETHEUS_URL=http://prometheus:9090/api/v1/write
+    volumes:
+      - ./configuration/alloy/config.alloy:/etc/alloy/config.alloy
```
Comment (FooRider) [inline sf-docker/sf-observability/docker-compose.yml:97]: Please include the config file using docker compose `configs`, also make sure to specify file attributes.

File: sf-docker/sf-observability/docker-compose.yml [hunk 10]
```diff
@@ -0,0 +1,214 @@
...
+    restart: unless-stopped
+  
+  prometheus:
+    extends:
+      service: common
+      file: docker-compose-common.yml
+    image: prom/prometheus:v2.47.0
+    profiles: [singlenode, multinode-aggregator]
+    volumes:
+      - ./configuration/prometheus:/etc/prometheus
```
Comment (FooRider) [inline sf-docker/sf-observability/docker-compose.yml:178]: Please include the config file using docker compose `configs`, also make sure to specify file attributes.

File: sf-docker/sf-observability/configuration/alloy/config.alloy (hunk 1 above)
Comment (egid-fertal-inno) [reply sf-docker/sf-observability/configuration/alloy/config.alloy:346]: removed

File: sf-docker/sf-observability/configuration/grafana/datasources/datasources.yml (hunk 2 above)
Comment (egid-fertal-inno) [reply sf-docker/sf-observability/configuration/grafana/datasources/datasources.yml:29]: I dont know :D

File: sf-docker/sf-observability/configuration/loki/loki-config.yaml (hunk 3 above)
Comment (egid-fertal-inno) [reply sf-docker/sf-observability/configuration/loki/loki-config.yaml:66]: it was left there by someone other then me :) i would rather leave it as it is

File: sf-docker/sf-observability/configuration/alloy/config.alloy (hunk 4 above)
Comment (egid-fertal-inno) [reply sf-docker/sf-observability/configuration/alloy/config.alloy:135]: yes, but it requires also changes made in dashboard, 
and we were having discussion that I should leave it for now :) that it is working and it is not perfect

Comment (egid-fertal-inno) [reply sf-docker/sf-observability/configuration/alloy/config.alloy:135]: of coruse I can do it, we just agree that I should not for now
should I do it ?

File: sf-docker/sf-observability/configuration/grafana/datasources/datasources.yml (hunk 7 above)
Comment (egid-fertal-inno) [reply sf-docker/sf-observability/configuration/grafana/datasources/datasources.yml:57]: removed

File: sf-docker/sf-observability/docker-compose.yml [hunk 11]
```diff
@@ -0,0 +1,214 @@
...
+volumes:
+  minio_data:
+  #obs_grafana_datasources:
+  #obs_grafana_dashboards:
+  grafana_data:
+  #obs_loki_config:
+  loki_data:
+  #obs_promtail_config:
+  #obs_prometheus_config:
+  prometheus_data:
```
Comment (FooRider) [inline sf-docker/sf-observability/docker-compose.yml:208]: You can remove unused volumes

File: sf-docker/sf-observability/docker-compose.yml (hunk 9 above)
Comment (egid-fertal-inno) [reply sf-docker/sf-observability/docker-compose.yml:97]: done

Comment (egid-fertal-inno) [reply sf-docker/sf-observability/docker-compose.yml:97]: done

File: sf-docker/sf-observability/docker-compose.yml [hunk 12]
```diff
@@ -0,0 +1,214 @@
...
+      file: docker-compose-common.yml
+    image: grafana/loki:2.9.1
+    profiles: [singlenode, multinode-aggregator]
+    depends_on:
+      obsminio:
+        condition: service_started
+      obsminio_createbuckets:
+        condition: service_completed_successfully
+    volumes:
+      - ./configuration/loki:/etc/loki
```
[the remark this answers is not in the document]
Comment (egid-fertal-inno) [reply sf-docker/sf-observability/docker-compose.yml:73]: done

File: sf-docker/sf-observability/docker-compose.yml (hunk 10 above)
Comment (egid-fertal-inno) [reply sf-docker/sf-observability/docker-compose.yml:178]: done

File: sf-docker/sf-observability/docker-compose.yml (hunk 8 above)
Comment (egid-fertal-inno) [reply sf-docker/sf-observability/docker-compose.yml:51]: I understand why it needs to be done, while solving other configs. But in this case
I need to do config for every file: Which are 7 files right now.

and then in every service I need to use long syntax
https://docs.docker.com/reference/compose-file/services/#long-syntax
to mount it in correct folder

are we sure we want do it this way ?:)

File: sf-docker/sf-observability/configuration/alloy/config.alloy (hunk 4 above)
Comment (FooRider) [reply sf-docker/sf-observability/configuration/alloy/config.alloy:135]: I'm sorry, I forgot about our yesterdays conversation.
But looking at it more soberly, I think we should remove it anyway.
The SFIT generated dashboards should still work and the other ones will be easy to fix.

File: sf-docker/sf-observability/configuration/grafana/datasources/datasources.yml (hunk 2 above)
Comment (FooRider) [reply sf-docker/sf-observability/configuration/grafana/datasources/datasources.yml:29]: Remove it, we can always add it later

**Review by FooRider — APPROVED** [review]

**Pull request merged**